### PR TITLE
Fix bugs with spatial attack

### PR DESCRIPTION
**Pull request by nottombrown**

### CODE REVIEW

File: cleverhans/attacks.py [hunk 1]
```diff
@@ -1742,7 +1742,6 @@ def generate(self, x, **kwargs):
         """
         Generate symbolic graph for adversarial examples and return.
         :param x: The model's symbolic inputs.
-        :param batch_size: (optional int) The size of batch during evaluation.
```
Comment (npapernot) [inline cleverhans/attacks.py]: Should batch_size also be removed from `feedable_kwargs` in that case?

File: cleverhans/attacks_tf.py [hunk 2]
```diff
@@ -1975,18 +1982,37 @@ def spm(x, model, batch_size=128, y=None, n_samples=None, dx_min=-0.1,
...
+    # import ipdb; ipdb.set_trace()
+
+    worst_sample_idx = tf.argmin(all_xents_samples_by_bs, axis=0) # B
+
+    out = tf.gather(all_adv_x, worst_sample_idx, axis=0)  # B x B x CHW
+
+    shp = tf.shape(out)
+    prnt_op = tf.Print(shp, [shp])
+    with tf.control_dependencies([tf.identity(prnt_op)]):
+
```
Comment (npapernot) [inline cleverhans/attacks_tf.py]: Extra blank line?

**Comment by npapernot** [comment]
LGTM once pep8 passes and small comments were addressed

**Comment by npapernot** [comment]
This will probably need a rebase after the switch to 2 spaces.

**Comment by nottombrown** [comment]
Closing in favor of #623

**Pull request closed without merging**